tests/e2e/tests/build/styles/extract-css.ts

```diff
@@ -1,21 +1,71 @@
+      'src/pre-rename-lazy-style.css': '.pre-rename-lazy-style { color: red }',
+      'src/common-entry-style.css': '.common-entry-style { color: red }',
+      'src/common-entry-script.js': 'console.log(\'common-entry-script\');'
+    }))
+    .then(() => updateJsonFile('angular-cli.json', configJson => {
+      const app = configJson['apps'][0];
+      app['styles'] = [
+        'url-style.css',
+        'string-style.css',
+        { input: 'input-style.css' },
+        { input: 'lazy-style.css', lazy: true },
+        { input: 'pre-rename-style.css', output: 'renamed-style' },
+        { input: 'pre-rename-lazy-style.css', output: 'renamed-lazy-style', lazy: true },
+        { input: 'common-entry-style.css', output: 'common-entry' }
+      ];
+      app['scripts'] = [{ input: 'common-entry-script.js', output: 'common-entry' }];
+    }))
     .then(() => ng('build', '--extract-css'))
-    .then(() => expectFileToMatch('dist/styles.bundle.css',
-      /div\s*{\s*background:\s*url\(more.svg\);\s*}/))
+    // files were created successfully
+    .then(() => expectFileToMatch('dist/url-style.bundle.css', 'url\(more.svg\)'))
+    .then(() => expectFileToMatch('dist/styles.bundle.css', '.string-style'))
+    .then(() => expectFileToMatch('dist/styles.bundle.css', '.input-style'))
+    .then(() => expectFileToMatch('dist/lazy-style.bundle.css', '.lazy-style'))
+    .then(() => expectFileToMatch('dist/renamed-style.bundle.css', '.pre-rename-style'))
+    .then(() => expectFileToMatch('dist/renamed-lazy-style.bundle.css', '.pre-rename-lazy-style'))
+    .then(() => expectFileToMatch('dist/common-entry.bundle.css', '.common-entry-style'))
+    .then(() => expectFileToMatch('dist/common-entry.bundle.js', 'common-entry-script'))
+    // there are no js entry points for css only bundles
+    .then(() => expectToFail(() => expectFileToExist('dist/style.bundle.js')))
+    .then(() => expectToFail(() => expectFileToExist('dist/lazy-style.bundle.js')))
+    .then(() => expectToFail(() => expectFileToExist('dist/renamed-style.bundle.js')))
+    .then(() => expectToFail(() => expectFileToExist('dist/renamed-lazy-style.bundle.js')))
+    // index.html lists the right bundles
+    .then(() => expectFileToMatch('dist/index.html', oneLineTrim`
+      <link href="renamed-style.bundle.css" rel="stylesheet"/>
+      <link href="styles.bundle.css" rel="stylesheet"/>
+      <link href="common-entry.bundle.css" rel="stylesheet"/>
+    `))
+    .then(() => expectFileToMatch('dist/index.html', oneLineTrim`
+      <script type="text/javascript" src="inline.bundle.js"></script>
+      <script type="text/javascript" src="vendor.bundle.js"></script>
+      <script type="text/javascript" src="common-entry.bundle.js"></script>
+      <script type="text/javascript" src="main.bundle.js"></script>
+    `))
+    // verify --deploy-url isn't applied to extracted urls
     .then(() => ng('build', '--extract-css', '--deploy-url=client/'))
```

Comment:
@filipesilva 
Sorry for keep editing the comment....
Not sure what the expect result right now. Will update after verifying later